Client/Frontend/Browser/TabTrayButtonExtensions.swift

```diff
@@ -5,20 +5,32 @@
 import UIKit
 
 extension UIButton {
-    static func privateModeButton() -> UIButton {
-        let privateTab = UIButton()
-        privateTab.setImage(UIImage.templateImageNamed("menu-NewPrivateTab-pbm"), forState: .Normal)
-        privateTab.tintColor = UIColor(white: 0.9, alpha: 1)
-        privateTab.setImage(UIImage(named: "menu-NewPrivateTab-pbm"), forState: .Highlighted)
-        privateTab.accessibilityLabel = NSLocalizedString("Private Tab", comment: "Accessibility label for the Private Tab button in the tab toolbar.")
-        return privateTab
+    class PrivateModeButton: ToggleButton {
```

Comment:
Nit: No need to declare this subclass inside the scope of the extension. I would move this outside.